buildSrc/build.gradle.kts

```diff
@@ -0,0 +1,107 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+plugins {
+    id("org.jlleitschuh.gradle.ktlint") version "9.2.1"
+    id("com.github.sherter.google-java-format") version "0.9"
+    `kotlin-dsl`
+}
+
+repositories {
+    mavenLocal()
+    maven(url = "https://maven.google.com/")
+    mavenCentral()
+    maven(url = "https://storage.googleapis.com/android-ci/mvn/")
+    maven(url = "https://plugins.gradle.org/m2/")
+}
+
+// Refer latest "perf-plugin" released version on https://maven.google.com/web/index.html?q=perf-plugin#com.google.firebase:perf-plugin
+// The System property allows us to integrate with an unreleased version from https://bityl.co/3oYt.
+// Refer go/fireperf-plugin-test-on-head for more details.
+val perfPluginVersion = System.getenv("FIREBASE_PERF_PLUGIN_VERSION") ?: "1.4.1"
+
+googleJavaFormat(Action {
+    toolVersion = "1.10.0"
+})
+
+dependencies {
+    // Firebase performance plugin, it should be added here because of how gradle dependency
+    // resolution works, otherwise it breaks Fireperf Test Apps.
+    // See https://github.com/gradle/gradle/issues/12286
+    implementation("com.google.firebase:perf-plugin:$perfPluginVersion")
+
+    implementation("com.google.auto.value:auto-value-annotations:1.8.1")
+    annotationProcessor("com.google.auto.value:auto-value:1.6.5")
+    implementation("digital.wup:android-maven-publish:3.6.3")
+    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.32")
```

Comment:
Since you're using the `kotlin-dsl`, this can actually be reduced to such:
```kotlin
implementation(kotlin("gradle-plugin", "1.4.32"))
```

But thats a design choice, especially since it differs from everything else.